src/Client/Connector.hpp

```diff
@@ -354,9 +360,22 @@ template<class BUFFER, class NetProvider>
 std::optional<Connection<BUFFER, NetProvider>>
 Connector<BUFFER, NetProvider>::waitAny(int timeout)
 {
+	if (m_Connections.empty()) {
+		LOG_DEBUG("waitAny() called on connector without connections");
+		return std::nullopt;
+	}
 	Timer timer{timeout};
 	timer.start();
 	while (m_ReadyToDecode.empty()) {
+		bool any_conn_has_no_err = false;
```

Comment:
Nit: wouldn't `has_alive_conn` be better? Too many logical negations with this name :)